test/unit_tests/cloud/publish_vitals.cpp

```diff
@@ -27,6 +27,9 @@
 #include "mock/mock_types.h"
 #include "system_publish_vitals.h"
 
+bool spark_cloud_flag_connected_called;
+int spark_cloud_flag_connected_result;
```

Comment:
There are test frameworks like hippomocks and fakeit that can help avoid the need to manually set return values and track calls. Not a call for change, just a heads up!

Reply:
This is handling open function calls, and not methods. This may be provided for by hippomocks, but I don't think this is/was available in fakeit at time of writing; I may be wrong.